cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -280,9 +283,27 @@ void DecisionTreeBase<T, L>::plant(
 
   total_temp_mem = tempmem->totalmem;
   MLCommon::TimerCPU timer;
-  grow_deep_tree(data, labels, rowids, n_sampled_rows, ncols,
-                 tree_params.max_features, dinfo.NLocalrows, sparsetree, treeid,
-                 tempmem);
+  if (tree_params.use_experimental_backend == true &&
+      tree_params.max_depth > 0 && tree_params.max_depth < 14 &&
```

Comment:
Why is `max_depth` limited to 14 for the experimental backend?

Reply:
`max_depth` of tree needs to be known a priori for the experimental backend due the way tree nodes are allocated. I have tested the backend for trees with depth less than 14. Plan is to change the node allocation strategy and then remove this condition.